File: dfs_and_bfs.cpp
#include<iostream>
#include<algorithm>
#include<queue>
#define NMAX 1000
#define MMAX NMAX*(NMAX-1)/2
using namespace std;

int n, m, i, j, x, y;
vector<int> adj[NMAX+1];
void bfs(int node){
    bool visited[n+1];
    for(i=1; i<=n; i++) visited[i] = false;
    queue<int> the_queue;
    the_queue.push(node);
    visited[node] = true;

    while( !the_queue.empty() ){
        for( auto itr: adj[the_queue.front()] )
            if( !visited[itr] ){
                the_queue.push(itr);
                visited[itr] = true;
            }
        cout<<the_queue.front()<<" ";
        the_queue.pop();
    }
}
void dfs_wrapped(int node, bool visited[]){
    cout<<node<<" ";
    for(auto itr : adj[node])
        if( !visited[itr] ){
            visited[itr] = true;
            dfs_wrapped(itr, visited);
        }
}
void dfs(int node){
    bool visited[n+1];
    for(i=1; i<=n; i++) visited[i] = false;
    visited[node] = true;
    dfs_wrapped(node, visited);
}
int main()
{
    cin >> n >> m;
    for(i=1; i<=m; i++){
        cin >> x >> y;
        adj[x].push_back(y);
        adj[y].push_back(x);
    }
    for(i=1; i<=n; i++) sort( adj[i].begin(), adj[i].end() );
    cout<<"LISTA DE ADIACENTA E:";
    for(i=1; i<=n; i++){
        cout<<"\n"<<i<<": ";
        for(auto itr:adj[i]) cout<<itr<<" ";
    }
    cout<<"\n";
    dfs(1);
    bfs(1);
}

File: dfs_and_bfs.py
from queue import Queue

class graph:
    def __init__(self, n:int):
        self.n = n
        self.vecini = {i:set({}) for i in range(n+1)}
        #self.visited = [ False for i in range(self.n + 1) ]
    def addEdge(self, x:int, y:int)->None:
        self.vecini[x].add(y)
        self.vecini[y].add(x)
    
    def DFS(self,node:int)->None:
        visited = [False for i in range(self.n + 1)]
        self.DFS_wrapped_function(node, visited)
    
    def DFS_wrapped_function(self, node: int, visited:list) -> None:
        visited[ node ] = True
        print(node, end=" ")
        for i in self.vecini[ node ]:
            if not visited[ i ]:
                self.DFS(i, visited)

    def BFS(self, node: int)->None:
        visited = [False for i in range(self.n + 1)]
        q = [node]
        visited[ node ] = True
        while( len(q) > 0 ):
            print(q[0], end=" ")
            for i in self.vecini[ q[0] ]:
                if not visited[ i ]:
                    q.append(i)
                    visited[i] = True
            q.pop(0)
            
    def liste_de_adiacenta(self):
        for i in range(self.n+1):
            print(i, end=": ")
            for j in self.vecini[i]:
                print(j, end=", ")
            print("")


if __name__ == "__main__":
    alfa = graph(8)
    alfa.addEdge(2, 4)
    alfa.addEdge(5, 3)
    alfa.addEdge(6, 2)
    alfa.addEdge(2, 5)
    alfa.liste_de_adiacenta()
    
    alfa.DFS(2)
    alfa.BFS(2)